Shopify/cli
# Add `<Dev>` component to app package

### WHY are these changes introduced?

`renderConcurrent` was becoming bloated and also quite clearly was starting to contain features that were only useful for the `dev` command. This meant some abstractions were leaking from the UI layer into the service layer.

### WHAT is this pull request doing?

I've introduced `ink` in the `app` package and created a new `Dev` component with a whole bunch of tests. Check out `Dev.test.tsx` to see them all!
I've also improved error handling by showing clear error messages in the UI and also aborting when one of the processes crashed, which is not something we were doing earlier.

### How to test your changes?

- Run `dev`
- Try enabling/disabling preview mode with `d`
- Try changing the state of preview mode in the partners dashboard, it should be reflected in the UI
- Stop dev, disable preview mode in the partners dashboard, start dev again and it should enable it automatically
- Press `q` and check that preview mode is being disabled automatically
- Do the same with `Ctrl+c`
